### PR TITLE
SS-29688: [MOL V3000 writer for Ketcher]: Writes R-Group Block

### PR DESCRIPTION
**Pull request by apar03**
- Adds MOL V3000 writer to Ketcher, which constructs the MOL V3000 representation from the underlying [`struct`](https://github.com/schrodinger/ketcher-schrodinger/blob/master/src/script/chem/struct/index.js) data structure.
- This PR addresses writing Reactions in V3000 format, as described in the [CTFile Representation Documentation, Chapter 10: The Extended Connection Table (V3000)](https://drive.google.com/file/d/0Bx3dsPc7eyZKdXlHTktQTFJUMHc/view)
- Tested with all sorts of RG Files
- Also added a test structure for IO testing, verified that it passes. The test molecule is shown below - 
image

Primary - @mycrobe

### CODE REVIEW

File: test/fixtures/v3000/r_group_block.mol [hunk 1]
```diff
@@ -0,0 +1,55 @@
+
```
Comment (mycrobe) [inline test/fixtures/v3000/r_group_block.mol:1]: Is this used in an automated test?

File: src/script/chem/molfile/molfile.js [hunk 2]
```diff
@@ -289,6 +290,26 @@ Molfile.prototype.writeCTab3000 = function (molecule) { // eslint-disable-line m
 	this.writeCR('M  V30 END CTAB');
 };
 
+Molfile.prototype.writeRGroups3000 = function (molecule) { // eslint-disable-line max-statements
```
Comment (mycrobe) [inline src/script/chem/molfile/molfile.js:295]: No JSDoc?

Comment (mycrobe) [reply src/script/chem/molfile/molfile.js:295]: JSDoc is on some functions but not others in this file. Is this intentional?

Comment (apar03) [reply src/script/chem/molfile/molfile.js:295]: Added.
Nope it's not intentional.. The existing V2000 writer methods are not documented, so we should be adding documentation to the new V3000 ones that we're adding, at the least.

File: test/fixtures/v3000/r_group_block.mol (hunk 1 above)
Comment (apar03) [reply test/fixtures/v3000/r_group_block.mol:1]: Yup, for an IO test - https://github.com/schrodinger/ketcher/blob/master/test/io/io.js, which 
- Constructs (reads or parses) the Ketcher Struct from the given test representation
- Writes back the representation using the writer
- Compares the output with the stored test representation

**Review by mycrobe — APPROVED** [review]

**Pull request merged**